app/helpers/announcements_helper.rb

```diff
@@ -4,10 +4,11 @@ def announcement_hidden?(announcement)
   end
 
   def current_announcement
+    return @current_announcement if defined? @current_announcement
     @current_announcement = Announcement.current
```

Comment:
Memoizing is what we need here, but this is not memoizing anything. Check your syntax against the docs.